jet/cdc/src/main/java/com/hazelcast/samples/jet/cdc/CdcRealTimeAnalysisDemo.java

```diff
@@ -0,0 +1,144 @@
+
+        HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
+        JetService jet = hz.getJet();
+        // trigger initialization
+        hz.getMap("CustomerStatsReport");
+
+        hz.getSql().execute(
+          """
+                  create mapping CustomerStatsReport (
+                    customerId int,
+                    customerFirstName varchar,
+                    customerLastName varchar,
+                    ordersTotal int,
+                    itemsTotal int,
+                    itemsAvg double)
+                  type IMap options (
+                    'keyFormat' = 'int',
+                    'valueFormat' = 'compact',
+                    'valueCompactTypeName'='CustomerStatsReport'
+                  );
+                  """
+        );
+
+        Pipeline pipeline = Pipeline.create().readFrom(
+                        postgres("inventory-db")
+                                .setDatabaseName("postgres")
+                                .setSchemaWhitelist("inventory")
+                                .setDatabaseAddress("localhost")
+                                .setDatabaseUser("postgres")
+                                .setDatabasePassword("cdcpwd1337")
+                                .setDatabasePort(5432)
+                                .setTableWhitelist("inventory.customers", "inventory.orders")
+                                .build())
+                .withoutTimestamps()
+                .flatMap(record -> {
+                    if (isOrder(record) && record.operation() == Operation.UPDATE) {
+                        return traverseItems(
+                                orderEvent(record.newValue().toObject(Order.class), INSERT),
+                                orderEvent(record.oldValue().toObject(Order.class), DELETE)
+                        );
+                    } else {
+                        return eventFor(record, record.operation());
+                    }
+                })
+                .groupingKey(ReportEvent::customerId)
+                .mapStateful(CustomerStatsReport::new, (state, key, record) -> {
+
+                    if (record.event() instanceof Customer customer) {
+                        state.setCustomerFirstName(customer.firstName());
+                        state.setCustomerLastName(customer.lastName());
+                        state.setCustomerId(customer.id());
+
+                    } else  {
+                        var order = (Order) record.event();
+                        var operation = record.operation();
+                        if (operation == Operation.SYNC || operation == INSERT) {
+                            state.setCustomerId(order.purchaser());
+                            state.setItemsTotal(state.getItemsTotal() + order.quantity());
+                            state.setOrdersTotal(state.getOrdersTotal() + 1);
+                            state.setItemsAvg(state.getItemsTotal() * 1.0d / state.getOrdersTotal());
```

Comment:
You can move this to the `CustomerStatsReport`?

Reply:
+1, looks much cleaner. Done